pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -278,7 +278,7 @@ func (s *Server) mcpController(
 	all := collections.Pilot.All()
 	cols := make([]sink.CollectionOptions, 0, len(all))
 	for _, c := range all {
-		cols = append(cols, sink.CollectionOptions{Name: c.Name().String(), Incremental: false})
+		cols = append(cols, sink.CollectionOptions{Name: c.Name().String(), Incremental: features.EnableIncrementalMCP})
```

Comment:
I understand we only enable incremental mcp for SE, not sure why all here?

Reply:
This implementation is not only for SE, but for all resources, such as VirtualService.

Reply:
IC, it depends on whether mcp server implemented

Reply:
According to the results of reading and testing the code, the mcp server does implement the increment, but the pilot does not support it in mcpController, so I try to implement it

Reply:
It makes less sense now, because galley is just a reference implementation for mcp server now.

Reply:
According to the design of istiod, the mcp server will be merged into istiod. In a multi-cluster scenario, the istiod of different clusters will synchronize data through mcp. I think that incremental updates are still necessary

Reply:
Nope, i think that is obsolete

Reply:
Does this mean that mcp has been abandoned?

Reply:
The MCP is still used in ConfigController. I think it is still necessary to support incremental MCP. The code is in "servicediscovery.go"
```go
func NewServiceDiscovery(configController model.ConfigStoreCache, store model.IstioConfigStore, xdsUpdater model.XDSUpdater) *ServiceEntryStore {
	c := &ServiceEntryStore{
		XdsUpdater:     xdsUpdater,
		store:          store,
		ip2instance:    map[string][]*model.ServiceInstance{},
		instances:      map[host.Name]map[string][]*model.ServiceInstance{},
		refreshIndexes: true,
	}
	if configController != nil {
		configController.RegisterEventHandler(serviceEntryKind,
			func(old, curr model.Config, event model.Event) {
				cs := convertServices(curr)
...
```